Description: Generate debug information on 32-bit architectures
Author: Nicholas Guriev <guriev-ns@ya.ru>
Last-Updated: Fri, 07 Feb 2020 18:27:32 +0300

--- a/cmake/options_linux.cmake
+++ b/cmake/options_linux.cmake
@@ -37,8 +37,10 @@ if (NOT DESKTOP_APP_USE_PACKAGED)
     )
 endif()
 if (build_linux32)
-    target_compile_options(common_options INTERFACE -g0)
-    target_link_options(common_options INTERFACE -g0)
+    # Less debugging information to avoid running out of address space when
+    # linking on architectures with reduced amount of memory.
+    target_compile_options(common_options INTERFACE -g1)
+    target_link_options(common_options INTERFACE -g1)
 elseif (NOT DESKTOP_APP_USE_PACKAGED)
     target_compile_options(common_options INTERFACE $<IF:$<CONFIG:Debug>,,-g -flto>)
     target_link_options(common_options INTERFACE $<IF:$<CONFIG:Debug>,,-g -flto -fuse-linker-plugin>)
